brian2/core/variables.py

```diff
@@ -274,6 +275,9 @@ def __repr__(self):
                                   constant=repr(self.constant),
                                   read_only=repr(self.read_only))
 
+    _state_tuple = property(lambda self: (self.dim, self.dtype, self.scalar,
+                                          self.constant, self.read_only,
+                                          self.dynamic, self.name))
```

Comment:
You mention this a bit later, but what about a system where all attributes are assumed to matter unless we explicitly exclude them? Like that, if we change the class in the future we are forced to make a decision about the new attributes. I feel like this isn't a lot more work and minimises the chance of silent errors in the future.